openmc/mgxs/mgxs.py

```diff
@@ -1884,13 +1886,19 @@ def tallies(self):
             energy = openmc.Filter('energy', group_edges)
             energyout = openmc.Filter('energyout', group_edges)
 
-            # Create a list of scores for each Tally to be created
-            if self.correction == 'P0':
-                scores = ['flux', 'scatter', 'scatter-P1']
-                filters = [[energy], [energy, energyout], [energyout]]
-            else:
-                scores = ['flux', 'scatter']
-                filters = [[energy], [energy, energyout]]
+            # Create lists of scores, filters for each Tally to be created
+            scores = ['flux']
+            filters = [[energy]]
+
+            # Create separate tallies for each moment
+            for moment in range(self.legendre_order+1):
+                scores.append('scatter-{}'.format(moment))
```

Comment:
Both ways will result in the same number of calls to `calc_pn`, but using the "scatter-Pn" scores will cut down on some of the overhead above that. Hard to say how much of a difference it would make, although it would be easy to test empirically.